services-custom/s3-transfer-manager/src/it/java/software/amazon/awssdk/transfer/s3/S3TransferManagerMultipartDownloadPauseResumeIntegrationTest.java

```diff
@@ -0,0 +1,109 @@
+import software.amazon.awssdk.transfer.s3.model.ResumableFileDownload;
+import software.amazon.awssdk.transfer.s3.progress.TransferProgressSnapshot;
+
+public class S3TransferManagerMultipartDownloadPauseResumeIntegrationTest extends S3IntegrationTestBase {
+    private static final String BUCKET = temporaryBucketName(S3TransferManagerMultipartDownloadPauseResumeIntegrationTest.class);
+    private static final String KEY = "key";
+
+    private static final long OBJ_SIZE = 32 * MB; // 32mib for 4 parts of 8 mib
+    private static File sourceFile;
+
+    @BeforeAll
+    public static void setup() throws Exception {
+        System.out.println("CREATING BUCKET");
+        createBucket(BUCKET);
+        sourceFile = new RandomTempFile(OBJ_SIZE);
+
+        // use async client for multipart upload (with default part size)
+        System.out.println("UPLOADING TEST OBJECT");
+        s3Async.putObject(PutObjectRequest.builder()
+                                          .bucket(BUCKET)
+                                          .key(KEY)
+                                          .build(), sourceFile.toPath())
+               .join();
+    }
+
+    @AfterAll
+    public static void cleanup() {
+        deleteBucketAndAllContents(BUCKET);
+        sourceFile.delete();
+    }
+
+    @Test
+    void pauseAndResume_shouldResumeDownload() {
+        Path path = RandomTempFile.randomUncreatedFile().toPath();
+        DownloadFileRequest request = DownloadFileRequest.builder()
+                                                         .getObjectRequest(b -> b.bucket(BUCKET).key(KEY))
+                                                         .destination(path)
+                                                         .build();
+        FileDownload download = tmJava.downloadFile(request);
+
+        // wait until we receive enough byte to stop somewhere between part 2 and 3, 18 Mib should do it
+        waitUntilAmountTransferred(download, 18 * MB);
+        ResumableFileDownload resumableFileDownload = download.pause();
+        FileDownload resumed = tmJava.resumeDownloadFile(resumableFileDownload);
+        resumed.completionFuture().join();
+        assertThat(path.toFile()).hasSameBinaryContentAs(sourceFile);
+    }
+
+    @Test
+    void pauseAndResume_whenAlreadyComplete_shouldHandleGracefully() {
+        Path path = RandomTempFile.randomUncreatedFile().toPath();
+        DownloadFileRequest request = DownloadFileRequest.builder()
+                                                         .getObjectRequest(b -> b.bucket(BUCKET).key(KEY))
+                                                         .destination(path)
+                                                         .build();
+        FileDownload download = tmJava.downloadFile(request);
+        download.completionFuture().join();
+        ResumableFileDownload resume = download.pause();
+        FileDownload resumedDownload = tmJava.resumeDownloadFile(resume);
```

Comment:
Since totalParts is not currently set, completedDownload() is never called. Can we assert that `"Sending downloadFileRequest "` logging is not called a second time, and the logging for a completedDownload is called?

Reply:
Good idea, I'll add some `LogCaptor` assertions